version: "3.3"
services:
  app:
    build: .
    ports:
      - '3000:3000'
    networks:
      - flask_network
    depends_on:
      - db
  db:
    image: 'postgres:latest'
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: password
      POSTGRES_DB: flask_db
    ports:
      - '5432:5432'
    restart: always
    volumes:
      - 'postgres_data:/var/lib/postgresql/data'
    networks:
      - flask_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${{POSTGRES_DB}} -t 1"]
      interval: 10s
      timeout: 10s
      retries: 10

  nginx:
    image: nginx:latest
    depends_on:
      - app
    volumes:
      - ./nginx:/etc/nginx/conf.d:ro 
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
    ports:
      - "80:80"
    networks:
      - flask_network
volumes:
  postgres_data: 
networks:
  flask_network:

